### PR TITLE
Json first

### PR DESCRIPTION
**Pull request by arpentnoir**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     },
     "license": "MIT",
     "dependencies": {
-        "js-yaml": "^3.13.1"
+        "js-yaml": "^3.13.1",
+        "json2yaml": "^1.1.0"
```
Comment (arpentnoir) [inline package.json]: why do you need both these packages?

File: src/association.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+          return mainPropertiesObj;
+     }
+     /**
+      * @function writeAssociationClassProperties
+      * @description adds property for association class
+      * @param {Object} main properties json object
+      * @param {UMLAssociationClassLink} associationClass 
+      */
+     writeAssociationClassProperties(mainPropertiesObj, associationClass) {
+          try {
```
Comment (arpentnoir) [inline src/association.js]: why do you have these big try blocks? what explicitly are you expecting to fail?

File: main.js [hunk 3]
```diff
@@ -32,10 +34,10 @@ function _handleGenerate(base, path, options) {
 /**
  * @function fileTypeSelection
  * @description Selects file type from the dropdown
- * @param {Object} base
+ * @param {UMLPackage} umlPackage
  * @param {Object} options
  */
-function fileTypeSelection(base, options) {
+function fileTypeSelection(umlPackage, options) {
      let filters = [
```
Comment (patilgirish) [inline main.js]: Do you still need this ?

File: main.js [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
+     // If umlPackage is not assigned, popup ElementPicker
+     if (!umlPackage) {
           app.elementPickerDialog
                .showDialog("Select the package or project to generate from", null, null) //type.UMLPackage
-               .then(function({
+               .then(function ({
                     buttonId,
                     returnValue
                }) {
                     if (buttonId === "ok") {
```
Comment (patilgirish) [inline main.js]: I think it's better to define an enum for buttonId, I see it's being used at lot of places.

**Pull request merged**